packages/datadog-plugin-azure-functions/test/integration-test/client.spec.js

```diff
@@ -0,0 +1,111 @@
+'use strict'
+
+const {
+  FakeAgent,
+  hookFile,
+  createSandbox,
+  curlAndAssertMessage
+} = require('../../../../integration-tests/helpers')
+const { spawn } = require('child_process')
+const { assert } = require('chai')
+const findProcess = require('find-process')
+const waitOn = require('wait-on')
+
+describe('esm', () => {
+  let agent
+  let proc
+  let sandbox
+
+  withVersions('azure-functions', '@azure/functions', version => {
+    before(async function () {
+      this.timeout(50000)
+      sandbox = await createSandbox([`@azure/functions@${version}`, 'azure-functions-core-tools@4'], false,
+        ['./packages/datadog-plugin-azure-functions/test/integration-test/fixtures/*'])
+    })
+
+    after(async function () {
+      this.timeout(50000)
+      await sandbox.remove()
+    })
+
+    beforeEach(async () => {
+      agent = await new FakeAgent().start()
+    })
+
+    afterEach(async () => {
+      const azureFuncProc = await findProcess('name', 'func', true)
```

Comment:
This shouldn't be needed if the process is started within our code as we should then be able to get the PID as it's started.

Reply:
Running the `func start` command actually spawns another process. This was the best method I found to find this spawned Azure Function process to kill it after each test. I also tried using [tree-kill](https://www.npmjs.com/package/tree-kill) but couldn't get it working when running the tests locally on my mac.

```
#! /usr/bin/env node

const path = require('path');
const spawn = require('child_process').spawn;
const args = process.argv;

function main() {
    const bin = path.resolve(path.join(path.dirname(__filename), '..', 'bin'));
    const funcProc = spawn(bin + '/func', args.slice(2), {
        stdio: [process.stdin, process.stdout, process.stderr, 'pipe']
    });

    funcProc.on('exit', code => {
        process.exit(code);
    });
}

main();
```

Reply:
It seems the process is actually `func`? Or are you saying calling `func` also spawns a different `func`?

Reply:
Correct, spawning `func` spawns a different `func`. The above code I shared is the code from Azure Functions Core Tools. So in the integration test `proc.pid` is a different value than `azureFuncProcPid`.

I'm happy to consider other methods that don't require adding another dev dependency! But this was the only reliable approach I could find while testing.

Reply:
It doesn't look detached, so wouldn't killing the parent process also kill the child process?

Reply:
It is attached, but for some reason I don't understand killing the parent process (PID 26515) does not kill the child processes. I'll do some more exploration to see if I can find a method to kill the child processes that avoids adding a new dev dependency.

```
USER            PID  PPID  PGID   SESS JOBC STAT   TT       TIME COMMAND
duncan.harvey 26491 11483 26491      0    1 S+   s001    0:00.15 node /opt/homebrew/Cellar/yarn/1.22.22/libexec/bin/yarn.js test:integration:plugins
duncan.harvey 26493 26491 26491      0    1 S+   s001    0:00.46 /Users/duncan.harvey/.nvm/versions/node/v20.10.0/bin/node /Users/duncan.harvey/go/src/github.com/DataDog/dd-trace-js/node_modules/.bin/mocha -r packages/dd-trace/test/setup/mocha.js packages/datadog-plugin-@(azure-functions)/test/integration-test/**/*.spec.js
duncan.harvey 26515 26493 26491      0    1 S+   s001    0:00.06 /Users/duncan.harvey/.nvm/versions/node/v20.10.0/bin/node /Users/duncan.harvey/go/src/github.com/DataDog/dd-trace-js/packages/datadog-plugin-azure-functions/test/integration-test/fixtures/node_modules/.bin/func start
duncan.harvey 26516 26515 26491      0    1 S+   s001    0:01.38 /Users/duncan.harvey/go/src/github.com/DataDog/dd-trace-js/packages/datadog-plugin-azure-functions/test/integration-test/fixtures/node_modules/azure-functions-core-tools/bin/func start
duncan.harvey 26589 26516 26491      0    1 S+   s001    0:00.34 /Users/duncan.harvey/.nvm/versions/node/v20.10.0/bin/node /Users/duncan.harvey/go/src/github.com/DataDog/dd-trace-js/packages/datadog-plugin-azure-functions/test/integration-test/fixtures/node_modules/azure-functions-core-tools/bin/workers/node/dist/src/nodejsWorker.js --host 127.0.0.1 --port 56210 --workerId 1a8f50d0-e274-47f3-af07-664cf84029b2 --requestId 3f7963a5-de29-481f-acc3-7cce7af17ede --grpcMaxMessageLength 2147483647 --functions-uri http://127.0.0.1:56210/ --functions-worker-id 1a8f50d0-e274-47f3-af07-664cf84029b2 --functions-request-id 3f7963a5-de29-481f-acc3-7cce7af17ede --functions-grpc-max-message-length 2147483647
```

Reply:
I was able to remove the `find-process` dependency by calling the nested `func` command directly. Now that I have that pid I can easily kill the process at the end of the test.